app/models/import_record_detail.rb

```diff
@@ -0,0 +1,124 @@
+class ImportRecordDetail < ActiveRecord::Base
```

Comment:
Question 1: This adds a great new feature for understanding what happens during individual import tasks, especially when something goes wrong.  An assumption I have is that the most immediately useful thing for improving logging would be to just see the data for the importer jobs in one place in the admin page.  The other would be to see the output from [print_initial_report](https://github.com/studentinsights/studentinsights/blob/master/lib/tasks/import.thor#L141) and `print_final_report`, which isn't included here.

I'm wondering about the benefits that adding this structure provides over writing to the `log` that's already passed and then persisting that (or some similarly minimal API).  The first benefit to this structure that I see is that we can show explicit columns in the admin panel, and we can sort them.  That doesn't seem like it adds much, are there other benefits I'm not thinking of?  It does feel like this is capturing some generally true things about records-based imports (records get processed, skipped, created, updated, deleted) that are unlikely to change.

A downside of adding more structure like this is that it's harder for me to see how we could add more logging for other things.  For example, we might want to track things like how many students are null when importing from STAR? or how many courses were nil for section records?  Here I think we'd have to decide about adding a column, but maybe there are other ways to enable this kind of logging.  A benefit for capturing logs, or just asking each importer to log directly or to return a hash of data is that import tasks can track whatever they want (or they can all track the same things too!).